server/src/main/java/io/littlehorse/server/listener/ServerListener.java

```diff
@@ -43,6 +43,8 @@ public void start() {
 
     @Override
     public void close() {
+        // This forcibly closes all grpc connections rather than waiting for them to
+        // complete. This cuts off all streaming connections too.
         server.shutdownNow();
```

Comment:
were you to change this right? to server.shutdown();

Reply:
`server.shutDown()` made the server hang because we would need to individually close each client request (especially the bidirectional stream ones with the task workers) before the server is allowed to shut down.

Also `server.isShutDown()` is basically just a flag that returns whether `server.shutDown()` was called...not really a health metric ):